psa-crypto-sys/build.rs

```diff
@@ -120,70 +122,74 @@ mod common {
-        let shim_bindings = bindgen::Builder::default()
+        // Common shim builder settings
+        let mut shim_builder = bindgen::Builder::default()
             .clang_arg(format!("-I{}", out_dir))
-            .clang_arg("-DMBEDTLS_CONFIG_FILE=<config.h>")
             .clang_arg(format!("-I{}", mbed_include_dir))
             .header("src/c/shim.h")
             .blocklist_type("max_align_t")
             .generate_comments(false)
-            .size_t_is_usize(true)
-            .generate()
-            .map_err(|_| {
-                Error::new(
-                    ErrorKind::Other,
-                    "Unable to generate bindings to mbed crypto",
-                )
-            })?;
+            .size_t_is_usize(true);
 
         #[cfg(feature = "prefix")]
-        let shim_bindings = bindgen::Builder::default()
-            .clang_arg(format!("-I{}", out_dir))
-            .clang_arg("-DMBEDTLS_CONFIG_FILE=<config.h>")
-            .clang_arg(format!("-I{}", mbed_include_dir))
-            .header("src/c/shim.h")
-            .blocklist_type("max_align_t")
-            .generate_comments(false)
-            .size_t_is_usize(true)
-            .parse_callbacks(Box::new(RenameCallbacks {}))
-            .generate()
-            .map_err(|_| {
-                Error::new(
-                    ErrorKind::Other,
-                    "Unable to generate bindings to mbed crypto",
-                )
-            })?;
+        {
+            shim_builder = shim_builder.parse_callbacks(Box::new(RenameCallbacks {}));
+        }
+
+        if !external_mbedtls {
+            shim_builder =
+                shim_builder.clang_arg(format!("-DMBEDTLS_CONFIG_FILE=\"{}\"", CONFIG_FILE));
+        }
+
+        // Build the bindings
+        let shim_bindings = shim_builder.generate().map_err(|_| {
+            Error::new(
+                ErrorKind::Other,
+                "Unable to generate bindings to mbed crypto",
+            )
+        })?;
 
         let out_path = PathBuf::from(env::var("OUT_DIR").unwrap());
         shim_bindings.write_to_file(out_path.join("shim_bindings.rs"))?;
 
         Ok(())
     }
 
-    pub fn compile_shim_library(include_dir: String, metadata: bool) -> Result<PathBuf> {
+    pub fn compile_shim_library(
+        include_dir: String,
+        metadata: bool,
+        external_mbedtls: bool,
+    ) -> Result<PathBuf> {
         let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
 
         let shimlib_name = "libmbedcryptoshim.a";
 
         // Compile and package the shim library
-        cc::Build::new()
-            .include(&out_dir)
-            .define("MBEDTLS_CONFIG_FILE", "<config.h>")
-            .include(include_dir)
-            .file("./src/c/shim.c")
-            .warnings(true)
-            .flag("-Werror")
-            .opt_level(2)
-            .cargo_metadata(metadata)
-            .try_compile(shimlib_name)
+        let mut cfg = cc::Build::new();
+        _ = cfg.include(&out_dir);
```

Comment:
nit: Why are we doing `_ = cfg.function()` instead of the `cfg.function1().function2()` pattern? Can we not leave it as it was?

Reply:
> nit: Why are we doing `_ = cfg.function()` instead of the `cfg.function1().function2()` pattern? Can we not leave it as it was?

I tried that, it's what I wanted to do, originally I couldn't find a way to make the borrow checker happy, but I just found a way. Here is was the error, change incoming.
```bash
error[E0716]: temporary value dropped while borrowed
   --> psa-crypto-sys/build.rs:179:23
```

Reply:
> nit: Why are we doing `_ = cfg.function()` instead of the `cfg.function1().function2()` pattern? Can we not leave it as it was?

I tried that, it's what I wanted to do, originally I couldn't find a way to make the borrow checker happy, but I just found a way. Here is was the error, change incoming.
```bash
error[E0716]: temporary value dropped while borrowed
   --> psa-crypto-sys/build.rs:179:23
```

Reply:
Thanks :)
There is an extra tab that is being flagged by the CI.